### ISSUE TITLE
sysconf(_SC_NPROCESSORS_CONF) get 0 cpu number

### ISSUE THREAD

**Issue by quanbaixin** [lxc/lxcfs#469]
sysconf(_SC_NPROCESSORS_CONF) is call get_nprocs_conf() to get cpu number, it read /sys/devices/system/cpu dir and count the number of directories named cpuX.
When I mount the lxcfs/sys/devices/system/cpu to /sys/devices/system/cpu，sys_readdir() in sys_fuse.c is worked，it return entris, but the entry->d_type is zero, not DT_DIR.

int get_nprocs_conf() {
  // On x86 kernels you can use /proc/cpuinfo for this, but on ARM kernels offline CPUs disappear
  // from there. This method works on both.

  ScopedReaddir reader("/sys/devices/system/cpu");
  if (reader.IsBad()) {
    return 1;
  }

  int result = 0;
  dirent* entry;
  while ((entry = reader.ReadEntry()) != nullptr) {
    if (entry->d_type == DT_DIR && __matches_cpuN(entry->d_name)) {
      ++result;
    }
  }
  return result;
}

**Comment by brauner** [comment]
Would you be up for sending a patch?

**Comment by pkasemir** [comment]
I had a similar problem with sysconf(_SC_NPROCESSORS_CONF) returning 0. My issue was fixed by:
https://github.com/lxc/lxcfs/commit/2e13f0be3ad903adf870e7a3032c3356c0c9e4eb

You can wait to try the next stable 5.0 release to see if it fixes your issue also.

**Comment by stgraber** [comment]
Yeah, I think we figured out what's going on with that logic.
There are FUSE3 versions (older ones) which will never set the right type, but for the rest, newer FUSE2 and FUSE3 will handle it fine and we have detection logic in our code to see if it will work or not.

I think we can consider this one closed now.

**Comment by jaskij** [comment]
I have lxcfs 5.0, and ran into this issue while building OpenBLAS: https://github.com/xianyi/OpenBLAS/issues/3636

```shell
$ pacman -Qs lxc
local/lxc 1:4.0.12-1
    Linux Containers
local/lxcfs 5.0.0-1
    FUSE filesystem for LXC
local/lxd 5.1-2
    Daemon based on liblxc offering a REST API to manage containers
```

Guest is `images:debian/bullseye`

**Comment by stgraber** [comment]
That would be because the fix was merged after 5.0

**Comment by C0rn3j** [comment]
Looks like this triggered for us on 2022-07-01 still (guessing from when app errors begun, tested for sure today)

Host itself is Ubuntu 20.04 with snap LXD running since June.

```
# uptime -s    
2022-06-02 11:01:40
```

Would that still be running 5.0 or less at the time or is this bug still a problem?
